### PR TITLE
Feat: Add interceptor

### LINKED ISSUE #32: Feat: if items_per_page equals -1, return all items

### PR DESCRIPTION
**Pull request by ZixMai**
1) Removed exception handle blocks, added interceptor instead.
2) Added permission denied error.
3) Added grpc-interceptor module to poetry files.
4) Modified pagination (if items_on_page equals -1 get all items).
5) Changed proto file because of interceptor.
6) Modified main.py by adding handler methods.
7) Added _connected flag and disconnect method to postgres class.

### CODE REVIEW

File: event_service/db/postgres_client.py [hunk 1]
```diff
@@ -16,6 +16,8 @@ class PostgresClient:
     db : prisma.Client
         Prisma db client.
 
+    _connected : bool
+
```
Comment (Nhsdkk) [inline event_service/db/postgres_client.py:21]: Docsting?

File: event_service/db/postgres_client.py [hunk 2]
```diff
@@ -24,6 +26,7 @@ class PostgresClient:
     """
```
Comment (Nhsdkk) [inline event_service/db/postgres_client.py:29]: Add docstring to disconnect method

File: event_service/repository/event_repository_impl.py [hunk 3]
```diff
@@ -141,18 +145,25 @@ async def get_events_by_events_ids(
...
-                "deleted_at": None,
-            },
-            skip=items_per_page * (page_number - 1),
-            take=items_per_page,
+        db_events: Optional[List[PrismaEvent]] = (
+            await self._db_client.db.event.find_many(
+                where={
+                    "id": {"in": [event_id for event_id in events_ids]},
+                    "deleted_at": None,
+                },
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Simplify

File: event_service/repository/event_repository_impl.py [hunk 4]
```diff
@@ -71,15 +71,19 @@ async def get_events_by_author_id(
...
-        db_events: Optional[
-            List[PrismaEvent]
-        ] = await self._db_client.db.event.find_many(
-            where={"id": author_id, "deleted_at": None},
-            skip=items_per_page * (page_number - 1),
-            take=items_per_page,
+        db_events: Optional[List[PrismaEvent]] = (
+            await self._db_client.db.event.find_many(
+                where={"id": author_id, "deleted_at": None},
+                skip=items_per_page * (page_number - 1),
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Simplify

File: event_service/repository/event_repository_impl.py [hunk 5]
```diff
@@ -176,21 +187,23 @@ async def get_all_events(
...
         """
-        db_events: Optional[
-            List[PrismaEvent]
-        ] = await self._db_client.db.event.find_many(
-            skip=items_per_page * (page_number - 1),
-            take=items_per_page,
+        db_events: Optional[List[PrismaEvent]] = (
+            await self._db_client.db.event.find_many(
+                skip=items_per_page * (page_number - 1),
+                take=items_per_page,
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Simplify

File: event_service/utils/custom_interceptor.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                the chain.
+        request_or_iterator: The RPC request, as a protobuf message if it is a
+            unary request, or an iterator of protobuf messages if it is a streaming
+            request.
+        context: The ServicerContext pass by gRPC to the service.
+        _: A string of the form "/protobuf.package.Service/Method"
+
+        Returns
+        -------
+        Returns the result of method(request_or_iterator)
```
Comment (Nhsdkk) [inline event_service/utils/custom_interceptor.py:49]: Add new line

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Review by MaxPlays35 — APPROVED** [review]
Approved

**Pull request merged**